firebase-firestore/src/main/java/com/google/firebase/firestore/remote/GrpcCallProvider.java

```diff
@@ -198,4 +201,84 @@ void shutdown() {
       Thread.currentThread().interrupt();
     }
   }
+
+  /**
+   * Monitors the connectivity state of the gRPC channel and resets the channel when gRPC fails to
+   * connect.
+   *
+   * <p>We currently cannot configure timeouts in connection attempts for gRPC
+   * (https://github.com/grpc/grpc-java/issues/1943), and until they support doing so, the gRPC
+   * connection can stay open for up to 2+ minutes before shutting down.
```

Comment:
Optional: this would be slightly clearer if we made the deficiency we're working around about the notification of shut down rather than the shut down itself. Suggestion: 

> ... open for up to 2+ minutes before notifying us that it has shut down.